### PR TITLE
 SRE-91 | Eliminate temporary files from avatar upload process 

### PR DESCRIPTION
**Pull request by mszabo-wikia**
Remove a cumbersome and slow process involving temporary files, upload straight to service instead.

https://wikia-inc.atlassian.net/browse/SRE-91

### CODE REVIEW

File: extensions/wikia/UserProfilePageV3/UserProfilePageHooks.class.php [hunk 1]
```diff
@@ -39,4 +39,19 @@ static public function onArticleSaveComplete(
...
+
+	/**
+	 * Exposes variables required by profile edit modal to JS
+	 * 
+	 * @param array $vars
+	 * @param OutputPage $out
+	 */
+	public static function onMakeGlobalVariablesScript( array &$vars, OutputPage $out ) {
+		if ( BodyController::showUserPagesHeader( $out->getTitle() ) ) {
+			$urlProvider = new \Wikia\Service\Gateway\KubernetesExternalUrlProvider();
```
Comment (rybmat) [inline extensions/wikia/UserProfilePageV3/UserProfilePageHooks.class.php]: import?

**Review by rybmat — APPROVED** [review]

File: extensions/wikia/UserProfilePageV3/templates/UserProfilePage_renderLightbox.php [hunk 2]
```diff
@@ -13,31 +13,25 @@
...
-							<input type="hidden" name="UPPLightboxDefaultAvatar" id="UPPLightboxDefaultAvatar" value="">
-							<input type="hidden" name="UPPLightboxFbAvatar" id="UPPLightboxFbAvatar" value="">
-							<input type="hidden" name="MAX_FILE_SIZE" value="<?= $avatarMaxSize; ?>">
-						</div>
-					<? endif; ?>
+					<div class="input-group">
+						<label for="avatar-upload-input">
+							<?= wfMessage( 'user-identity-box-avatar-upload-avatar' )->escaped(); ?>
+						</label>
+						<input type="file" name="file" id="avatar-upload-input" accept="image/*" />
```
Comment (jsutterfield) [inline extensions/wikia/UserProfilePageV3/templates/UserProfilePage_renderLightbox.php:25]: We want to limit the types of files that can be uploaded to [these](https://github.com/Wikia/app/pull/16081/files?utf8=%E2%9C%93&diff=split#diff-24765cebf3960a2bf8cb16081c7af852L24), right? Maybe we could enumerate those here in the `accept` attribute?

File: extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php [hunk 3]
```diff
@@ -724,16 +436,11 @@ private function renderAvatarLightbox( $userId ) {
...
 		$this->setVal( 'defaultAvatars', $this->getDefaultAvatars() );
-		$this->setVal( 'isUploadsPossible',
-			$this->wg->EnableUploads &&
-			$this->wg->User->isAllowed( 'upload' ) &&
-			empty( $this->wg->AvatarsMaintenance )
-		);
 
 		$this->setVal( 'avatarName', $user->getGlobalAttribute( AVATAR_USER_OPTION_NAME ) );
 		$this->setVal( 'userId', $userId );
 		$this->setVal( 'avatarMaxSize', self::AVATAR_MAX_SIZE );
```
Comment (jsutterfield) [inline extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php]: I don't think this is being used anymore, right? AFAICT the only usage was [here](https://github.com/Wikia/app/pull/16081/files?utf8=%E2%9C%93&diff=split#diff-fa16031792b15f1939e781da1d596a33L30).

File: extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php [hunk 4]
```diff
@@ -724,16 +436,11 @@ private function renderAvatarLightbox( $userId ) {
...
 
 		$this->setVal( 'defaultAvatars', $this->getDefaultAvatars() );
-		$this->setVal( 'isUploadsPossible',
-			$this->wg->EnableUploads &&
-			$this->wg->User->isAllowed( 'upload' ) &&
-			empty( $this->wg->AvatarsMaintenance )
-		);
 
 		$this->setVal( 'avatarName', $user->getGlobalAttribute( AVATAR_USER_OPTION_NAME ) );
 		$this->setVal( 'userId', $userId );
```
Comment (jsutterfield) [inline extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php]: Same with userId and avatarName, looks like they're not being used anymore?

**Review by jsutterfield — APPROVED** [review]
Looks great! Just a couple minor comments.

File: extensions/wikia/UserProfilePageV3/templates/UserProfilePage_renderLightbox.php (hunk 2 above)
Comment (mszabo-wikia) [reply extensions/wikia/UserProfilePageV3/templates/UserProfilePage_renderLightbox.php:25]: Yeah I was thinking about that, but ultimately the source of truth for that is the avatar service. So if we started listing them here we'd have to remember to keep them in sync.

Comment (jsutterfield) [reply extensions/wikia/UserProfilePageV3/templates/UserProfilePage_renderLightbox.php:25]: Yea, that's a fair point. This approach makes sense to me 👍

File: extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php (hunk 4 above)
Comment (mszabo-wikia) [reply extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php]: Thanks! Removed in 5fca599

File: extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php (hunk 3 above)
Comment (mszabo-wikia) [reply extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php]: Yeah :) Removed in 5fca599

File: extensions/wikia/UserProfilePageV3/UserProfilePageHooks.class.php (hunk 1 above)
Comment (mszabo-wikia) [reply extensions/wikia/UserProfilePageV3/UserProfilePageHooks.class.php]: Addressed in 5fca599

**Pull request merged**